test/network/network_test.cpp

```diff
@@ -1,3 +1,5 @@
+#pragma GCC diagnostic ignored "-Wconversion"
```

Comment:
This shouldn't be necessary. If you were getting warnings about implicit conversation, those are typically resolved with an explicit static_cast.

Reply:
GCC still seems to complain about that static_cast from size_t (uint64_t) to int32_t due to information loss though.

Reply:
Take another look. It seemed to work for me in suppressing the warnings, but I only did a sample.

Reply:
Ya I did it in the end. My bad